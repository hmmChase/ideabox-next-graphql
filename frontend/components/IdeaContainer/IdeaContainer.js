import { gql } from 'apollo-boost';
import { Query } from 'react-apollo';
import styled from 'styled-components';
import IdeaCard from '../IdeaCard/IdeaCard';

export const ALL_IDEAS_QUERY = gql`
  query ALL_IDEAS_QUERY {
    ideas {
      id
      idea
    }
  }
`;

class IdeaContainer extends React.PureComponent {
  displayIdeaCards = data => data.ideas.map(idea => <IdeaCard key={`ideaCard${idea.id}`} {...idea} />);

  render() {
    return (
      <Query query={ALL_IDEAS_QUERY}>
        {({ data, error, loading }) => {
          if (loading) return <p>Loading...</p>;
          if (error) return <p>Error: {error.message}</p>;
          return <StyledUl>{this.displayIdeaCards(data)}</StyledUl>;
        }}
      </Query>
    );
  }
}

export default IdeaContainer;

const StyledUl = styled.ul`
  width: 100%;
  max-width: 1000px;
  display: flex;
  flex-direction: column;
  align-items: center;
  list-style: none;
  padding: 0;
  margin: 0 auto;
`;
