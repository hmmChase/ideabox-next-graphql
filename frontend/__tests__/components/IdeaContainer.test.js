import React from 'react';
import { shallow } from 'enzyme';
import IdeaContainer from '../../components/IdeaContainer/IdeaContainer';

describe('IdeaContainer', () => {
  let mockProps;
  let wrapper;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = { children: [] };
    wrapper = shallow(<IdeaContainer {...mockProps} />, {
      disableLifecycleMethods: true
    });
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
