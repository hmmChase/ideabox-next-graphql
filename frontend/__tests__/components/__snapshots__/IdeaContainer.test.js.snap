// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaContainer matches snapshot 1`] = `
<Query
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "ALL_IDEAS_QUERY",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "ideas",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "idea",
                      },
                      "selectionSet": undefined,
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 69,
        "start": 0,
      },
    }
  }
>
  <Component />
</Query>
`;
